crates/polars-expr/src/expressions/window.rs

```diff
@@ -334,13 +333,8 @@ impl WindowExpr {
             // no explicit aggregations, map over the groups
             //`(col("x").sum() * col("y")).over("groups")`
             (WindowMapping::GroupsToRows, AggState::AggregatedList(_)) => {
-                if sorted_keys {
-                    if let GroupsProxy::Idx(g) = gb.get_groups() {
-                        debug_assert!(g.is_sorted_flag())
```

Comment:
`GroupsProxy::Idx` must be explicitly mapped back to the row positions regardless of whether it was sorted